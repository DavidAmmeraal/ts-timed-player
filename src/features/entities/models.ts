export interface Entity {
    id: string;
}

export interface EntitiesContainer {
    byId: {
        [key:string]: Entity
    },
    ids: string[],
}

const defaultEntitiesContainerProps: EntitiesContainer = {
    byId: {},
    ids: []
}

export const createEntitiesContainer = () => ({
    ...defaultEntitiesContainerProps,
});