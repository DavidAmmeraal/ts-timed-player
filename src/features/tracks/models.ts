import { Entity } from 'util/entities';

export interface ITrack extends Entity {
    start: Date,
    end: Date,
}