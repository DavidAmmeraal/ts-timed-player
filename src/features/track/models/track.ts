
interface Track {
    id: string,
    start: Date,
    end: Date
}