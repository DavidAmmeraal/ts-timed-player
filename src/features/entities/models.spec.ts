import { createEntitiesContainer } from './models';

describe('createEntitiesContainer()', () => {
    it('should create a new empty entities container.', () => {
        const container = createEntitiesContainer();
        expect(container).toMatchSnapshot();
    })
})