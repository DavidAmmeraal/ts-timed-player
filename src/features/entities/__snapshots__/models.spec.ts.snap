// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createEntitiesContainer() should create a new empty entities container. 1`] = `
Object {
  "byId": Object {},
  "ids": Array [],
}
`;
